pkg/credentialprovider/provider.go

```diff
@@ -107,3 +98,18 @@ func (d *CachingDockerConfigProvider) Provide(image string) DockerConfig {
 	}
 	return config
 }
+
+// NewDefaultDockerKeyring creates a DockerKeyring to use for resolving credentials,
+// which returns the default credentials from the .dockercfg file.
+func NewDefaultDockerKeyring() DockerKeyring {
+	keyring := &providersDockerKeyring{
+		Providers: make([]DockerConfigProvider, 0),
+	}
+
+	keyring.Providers = append(keyring.Providers, &CachingDockerConfigProvider{
```

Comment:
nit, drop the append and just build the slice manually since it will always have one entry.